subworkflows/local/prepare_genome.nf

```diff
@@ -16,7 +17,22 @@ workflow PREPARE_GENOME {
     main:
     ch_versions = Channel.empty()
 
-    ch_fasta = Channel.fromPath(fasta)
+    // MapSplice cannot deal with extra field in the fasta headers
+    // this removes all additional fields in the headers of the input fasta file
+    if( params.tool.contains('mapsplice') && params.module.contains('circrna_discovery') ) {
+
+        CLEAN_FASTA(Channel.value([[id: "${fasta.baseName}" + ".clean_headers" ], fasta]), [])
```

Comment:
You could get rid of the new Channel creation here, if you would use the `fasta_tuple` channel which is defined a few lines below as input

Reply:
`CLEAN_FASTA(fasta_tuple.map{meta, fasta -> [id: "${fasta.baseName}" + ".clean_headers"], fasta}, [])`

would be what I mean but probably won't make a difference, I missed the special naming at first glance

Reply:
thanks for explaining! I'm still not very used to groovy :)